crm_compassion/models/weekly_demand.py

```diff
@@ -99,61 +113,103 @@ def _compute_demand_events(self):
         for week in self:
             week.total_demand = week.number_children_website + \
+                week.number_sub_sponsorship + \
                 week.number_children_ambassador + week.number_children_events
 
     @api.model
     def _default_unsponsored_web(self):
         """ Compute average of unsponsored children on the website since
         one year.
         """
-        start_date = datetime.today() - timedelta(weeks=52)
+        start_date = datetime.today() - timedelta(weeks=STATS_DURATION)
         web_sponsored = self.env['recurring.contract'].search_count([
             ('channel', '=', 'internet'),
             ('start_date', '>=', fields.Date.to_string(start_date))
         ])
         allocate_per_week = int(self.env['ir.config_parameter'].get_param(
             'crm_compassion.number_children_web'))
-        return allocate_per_week - (float(web_sponsored) / 52.0)
+        return allocate_per_week - (float(web_sponsored) / STATS_DURATION)
 
     @api.model
     def _default_unsponsored_ambassador(self):
         """ Compute average of unsponsored children from ambassadors since
         one year.
         """
-        start_date = datetime.today() - timedelta(weeks=52)
+        start_date = datetime.today() - timedelta(weeks=STATS_DURATION)
         ambass_sponsored = self.env['recurring.contract'].search_count([
             ('origin_id.type', '=', 'partner'),
             ('origin_id.partner_id', '!=', False),
             ('start_date', '>=', fields.Date.to_string(start_date))
         ])
         allocate_per_week = int(self.env['ir.config_parameter'].get_param(
             'crm_compassion.number_children_ambassador'))
-        return allocate_per_week - (float(ambass_sponsored) / 52.0)
+        return allocate_per_week - (float(ambass_sponsored) / STATS_DURATION)
+
+    @api.depends('week_start_date')
+    @api.multi
+    def _compute_resupply_sub(self):
+        """ Compute SUB resupply. """
+        sub_average = self._default_demand_sub()
+        today = datetime.today()
+        rejected_sub = self.env['recurring.contract'].search([
+            ('origin_id.type', '=', 'sub'),
+            ('start_date', '>=',
+             datetime.today() - timedelta(weeks=STATS_DURATION)),
+            ('end_date', '!=', None)
+        ]).filtered(
+            lambda s: ((
+                           fields.Date.from_string(s.end_date) -
+                           fields.Date.from_string(s.start_date)
+                       ).days <= SUB_DURATION)
+        )
+        sub_reject_average = len(rejected_sub) / (SUB_DURATION / 7)
+        for week in self:
+            start_date = fields.Datetime.from_string(
+                week.week_start_date) - timedelta(days=SUB_DURATION)
+            if start_date <= today:
+                sub = self.env['recurring.contract'].search_count([
+                    ('origin_id.type', '=', 'sub'),
+                    ('start_date', '>=', fields.Date.from_string(
```

Comment:
why not using the variable `start_date`?
you should as well add in the search filter that then start_date must be <= than 90 days before the week end date.